### DISCUSSION TITLE
permission denied issue for installing software in step three： Configuring the shell

### DISCUSSION THREAD

**Discussion by Yuzhou-Arial** [biostars/biostar-handbook-issues#332] [Q&A] [answered]
i set the system for MacOS 14.5 followd by tap in "curl http://data.biostarhandbook.com/install.sh | bash" in the terminal. It gets error "permission denied" in step 3，like picture as follows. Does anyone knows why it happens and how to solve it ? 
<img width="1382" alt="截屏2024-05-21 12 55 27" src="https://github.com/biostars/biostar-handbook/assets/165237300/85db5664-b0f1-4723-b690-dd7bc1731f1b">

**Comment by ialbert** [comment, marked as answer]
this sounds like the `.bash_profile` is not writeable, quite unexpected and normally that should not be set up like that

maybe some other tool you installed before did that and it seems you have ran that tool as `sudo` so now your `.bash_profile` is owned by root and not writeable, 

you will need to make the file writeable, 

try this:

    sudo chmod a+w .bash_profile

you will need to enter your password for the command to run

**Reply by Yuzhou-Arial** [reply]
It works!!! Thank you~
